src/overlays/actors/ovl_En_Dnk/z_en_dnk.c

```diff
@@ -70,44 +133,336 @@ static DamageTable D_80A521FC = {
+                SkelAnime_Init(globalCtx, &this->skelAnime, &D_060023B8, NULL, this->jointTable, this->morphTable, 10);
+                func_80A514F0(&this->skelAnime, 18);
+                break;
+
+            case ENDNK_PARAMS_3_2:
+                SkelAnime_Init(globalCtx, &this->skelAnime, &D_06002468, NULL, this->jointTable, this->morphTable, 10);
+                func_80A514F0(&this->skelAnime, 35);
+                break;
+        }
+
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+        if (ENDNK_PARAMS_3C(&this->actor) == ENDNK_PARAMS_3C_4) {
+            this->actor.flags &= ~1;
+            this->actor.flags |= (0x10 | 0x20);
+            this->actionFunc = func_80A51890;
+            Actor_SetScale(&this->actor, 0.1f);
+        } else {
+            this->actor.flags &= ~1;
+            this->actionFunc = func_80A518DC;
+            Actor_SetScale(&this->actor, 0.01f);
+        }
+    }
+}
+
+void func_80A51890(EnDnk* this, GlobalContext* globalCtx) {
+    if (func_800EE29C(globalCtx, 0x7E)) {
+        func_800EDF24(&this->actor, globalCtx, func_800EE200(globalCtx, 0x7E));
+    }
+}
+
+void func_80A518DC(EnDnk* this, GlobalContext* globalCtx) {
+}
+
+void EnDnk_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDnk* this = THIS;
+
+    this->unk_28E = -1;
+    switch (ENDNK_PARAMS_3(&this->actor)) {
+        case ENDNK_PARAMS_3_1:
+            this->unk_28E = func_8013D924(0x12B, globalCtx);
+            break;
+
+        case ENDNK_PARAMS_3_0:
+            this->unk_28E = func_8013D924(0x135, globalCtx);
+            break;
+
+        case ENDNK_PARAMS_3_2:
+            this->unk_28E = func_8013D924(0x40, globalCtx);
+            break;
+    }
+
+    if (this->unk_28E >= 0) {
+        this->actionFunc = func_80A51648;
+    } else {
+        Actor_MarkForDeath(&this->actor);
+    }
+
+    this->unk_2A2 = D_80A521A0;
+    D_80A521A0++;
+}
+
+void EnDnk_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDnk* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnDnk_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDnk* this = THIS;
+    s32 pad;
+
+    this->actionFunc(this, globalCtx);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80A515C4(this);
+    Actor_SetHeight(&this->actor, 34.0f);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    func_80A52134(this, globalCtx);
+}
+
+s32 func_80A51A78(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnDnk* this = THIS;
+
+    this->unk_260[limbIndex] = *dList;
+    *dList = NULL;
+    return 0;
```

Comment:
```suggestion
    return false;
```
Since this actor has several sets of Draws, even I'm going to say hold of on naming the LimbDraw stuff until we actually know which case is which.

Reply:
Is that really a false? I left this because it seems more like a separate system, like it would be some enum that's actually returning OK, how it's most common to use 0 for no error, and anything non-zero as an error.

Reply:
We usually do them as false: if memory serves, if this function returns true the limb is not to be drawn (it's used at least once like that in OoT). It is actually a boolean, albeit one that is almost always false.

Reply:
https://github.com/zeldaret/mm/blob/master/src/code/z_skelanime.c#L334
Returning `true` means this functions has "overrided" drawing the original limb, so the original dlist it won't be drawn.